compiler/rustc_builtin_macros/src/deriving/debug.rs

```diff
@@ -94,12 +94,17 @@ fn show_substructure(cx: &mut ExtCtxt<'_>, span: Span, substr: &Substructure<'_>
         args.extend([fmt, name]);
         for i in 0..fields.len() {
             let field = &fields[i];
-            if is_struct {
-                let name = cx.expr_str(field.span, field.name.unwrap().name);
+            if let Some(name) = field.name {
```

Comment:
Can you revert this change? There is a comment above "Struct and tuples are similar enough that we use the same code for both, with some extra pieces for structs due to the field names." The `is_struct` bool currently distinguishes those extra pieces, and this change makes that less clear.